cmd/argocd/commands/root.go

```diff
@@ -18,6 +24,88 @@ import (
 	"github.com/argoproj/argo-cd/v2/util/localconfig"
 )
 
+type ArgoCDCLIOptions struct {
+	PluginHandler PluginHandler
+	Arguments     []string
+}
+
+// PluginHandler parses command line arguments
+// and performs executable filename lookups to search
+// for valid plugin files, and execute found plugins.
+type PluginHandler interface {
+	// LookForPlugin will iterate over a list of given prefixes
+	// in order to recognize valid plugin filenames.
+	// The first filepath to match a prefix is returned.
+	LookForPlugin(filename string) (string, bool)
+	// ExecutePlugin receives an executable's filepath, a slice
+	// of arguments, and a slice of environment variables
+	// to relay to the executable.
+	ExecutePlugin(executablePath string, cmdArgs, environment []string) error
+}
+
+// DefaultPluginHandler implements the PluginHandler interface
+type DefaultPluginHandler struct {
+	ValidPrefixes []string
+}
+
+func NewDefaultPluginHandler(validPrefixes []string) *DefaultPluginHandler {
+	return &DefaultPluginHandler{
+		ValidPrefixes: validPrefixes,
+	}
+}
+
+// LookForPlugin implements PluginHandler
+func (h *DefaultPluginHandler) LookForPlugin(filename string) (string, bool) {
+	for _, prefix := range h.ValidPrefixes {
+		path, err := exec.LookPath(fmt.Sprintf("%s-%s", prefix, filename))
+		if shouldSkipOnLookPathErr(err) || len(path) == 0 {
+			continue
+		}
+		return path, true
+	}
+	return "", false
+}
+
+// ExecutePlugin implements PluginHandler
+func (h *DefaultPluginHandler) ExecutePlugin(executablePath string, cmdArgs, environment []string) error {
+	// Windows does not support exec syscall.
```

Comment:
Windows doesn't support syscall but all OSs support `cmd.Run()`.
Why not having the plugin execution handled by a `Command` in all OSs and simplifying this code?

Reply:
PTAL now.